### PR TITLE
Use the interface for regression tests

### PR DESCRIPTION
**Pull request by henrykironde**

### CODE REVIEW

**Comment by ShivamNegi** [comment]
I think the PR works well. It is ready to go in. @ethanwhite what are your views?

File: test/test_regression.py [hunk 1]
```diff
@@ -27,6 +36,7 @@
 retriever_root_dir = os.path.abspath(os.path.join(file_location, os.pardir))
 working_script_dir = os.path.abspath(os.path.join(retriever_root_dir, "scripts"))
 HOMEDIR = os.path.expanduser('~')
+script_home = "~/.retriever/scripts"
```
Comment (ethanwhite) [inline test/test_regression.py]: Is there a reason to use `~` here instead of `HOMEDIR`?

File: test/test_regression.py [hunk 2]
```diff
@@ -47,21 +57,24 @@ def get_script_module(script_name):
     return imp.load_module(script_name, file, pathname, desc)
 
 
-def get_csv_md5(dataset, engines, tmpdir):
+def get_csv_md5(dataset, engines, tmpdir, install_function, cofig):
```
Comment (ethanwhite) [inline test/test_regression.py]: Is `cofig` is supposed to be `config` as in "configuration"?

File: test/test_regression.py [hunk 3]
```diff
@@ -47,21 +57,24 @@ def get_script_module(script_name):
...
     workdir = tmpdir.mkdtemp()
     workdir.chdir()
+    install_function(dataset.replace("_", "-"), **cofig)
```
Comment (ethanwhite) [inline test/test_regression.py]: If changing `cofig` -> `config` do it here as well.

File: test/test_regression.py (hunk 2 above)
Comment (ethanwhite) [inline test/test_regression.py]: It looks like we're only passing in one engine at a time in which case change `engines` to `engine`.

File: test/test_regression.py (hunk 1 above)
Comment (henrykironde) [inline test/test_regression.py]: I didn't see that. will make changes

**Comment by henrykironde** [comment]
@ethanwhite updated

**Pull request merged**